core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala

```diff
@@ -1364,6 +1365,42 @@ class WorkspaceServiceSpec extends AnyFlatSpec with ScalatestRouteTest with Matc
     verify(services.resourceBufferService).getGoogleProjectFromBuffer(any[ProjectPoolType], any[String])
   }
 
+  it should "Update a Google Project name after claiming a project from Resource Buffering Service" in withTestDataServices { services =>
+    val newWorkspaceNamespace = testData.testProject1Name.value
+    val newWorkspaceName = "space_for_workin"
+    val workspaceRequest = WorkspaceRequest(newWorkspaceNamespace, newWorkspaceName, Map.empty)
+    val captor = ArgumentCaptor.forClass(classOf[Project])
+
+    val workspace = Await.result(services.workspaceService.createWorkspace(workspaceRequest), Duration.Inf)
+
+    verify(services.gcsDAO).updateGoogleProject(ArgumentMatchers.eq(GoogleProjectId("project-from-buffer")), captor.capture())
+    val capturedProject = captor.getValue.asInstanceOf[Project] // Explicit cast needed since Scala type interference and capturing parameters with Mockito don't play nicely together here
+
+    val expectedProjectName = "arbitrary--space-for-workin"
+    val actualProjectName = capturedProject.getName
+
+    actualProjectName shouldBe expectedProjectName
+  }
+
+  it should "Apply labels to a Google Project after claiming a project from Resource Buffering Service" in withTestDataServices { services =>
+    val newWorkspaceNamespace = testData.testProject1Name.value
+    val newWorkspaceName = "space_for_workin"
+    val workspaceRequest = WorkspaceRequest(newWorkspaceNamespace, newWorkspaceName, Map.empty)
+    val captor = ArgumentCaptor.forClass(classOf[Project])
+
+    val workspace = Await.result(services.workspaceService.createWorkspace(workspaceRequest), Duration.Inf)
+
+    verify(services.gcsDAO).updateGoogleProject(ArgumentMatchers.eq(GoogleProjectId("project-from-buffer")), captor.capture())
+    val capturedProject = captor.getValue.asInstanceOf[Project] // Explicit cast needed since Scala type interference and capturing parameters with Mockito don't play nicely together here
+
+    val expectedNewLabels = services.gcsDAO.labelSafeMap(Map("workspaceNamespace" -> newWorkspaceNamespace, "workspaceName" -> newWorkspaceName, "workspaceId" -> workspace.workspaceId), "")
```

Comment:
I would just hard code the expected values here rather than using `labelSafeMap`.  I would also give the workspace name a couple of special characters so that you can make sure the `createWorkspace` call creates the labels with the appropriate special character replacement.  Maybe even make a long workspace name too to make sure it truncates properly.